Lib/modulefinder.py

```diff
@@ -43,6 +51,67 @@ def ReplacePackage(oldname, newname):
     replacePackageMap[oldname] = newname
 
 
+def _find_module(name, path=None):
+    """An importlib reimplementation of imp.find_module (for our purposes)."""
+
+    importlib.machinery.PathFinder.invalidate_caches()
+
+    spec = importlib.machinery.PathFinder.find_spec(name, path+sys.path)
+
+    if spec is None:
+        raise ImportError("No module named {name!r}".format(name=name), name=name)
+
+    if isinstance(spec.loader, type):  # Some special cases:
+
+        if issubclass(spec.loader, importlib.machinery.BuiltinImporter):
+            return None, None, ("", "", _C_BUILTIN)
+
+        if issubclass(spec.loader, importlib.machinery.FrozenImporter):
+            return None, None, ("", "", _PY_FROZEN)
+
+    file_path = spec.origin
+
+    try:
+
+        # `imp` behavior is s bit funny here:
+
+        # If the path is a descendant of the CWD, it is reported as a relative filepath
+        # ...unless it is an extension, in which case it will be absolute (see below)
+        # ...otherwise, it is fully resolved.
```

Comment:
I wonder if this behaviour should be emulated, although there's bound to be someone that relies on the current behaviour.

Reply:
Good news: it looks like `PathFinder` does this weird path formatting for us! So either way, no need for this ugly code.